cpu/sam0_common/periph/uart.c

```diff
@@ -136,14 +136,14 @@ void uart_poweroff(uart_t uart)
 
 static inline void irq_handler(unsigned uartnum)
 {
-    if (dev(uartnum)->INTFLAG.reg & SERCOM_USART_INTFLAG_RXC) {
+    if (dev(uartnum)->INTFLAG.bit.RXC) {
         /* interrupt flag is cleared by reading the data register */
         uart_ctx[uartnum].rx_cb(uart_ctx[uartnum].arg,
                                 (uint8_t)(dev(uartnum)->DATA.reg));
     }
-    else if (dev(uartnum)->INTFLAG.reg & SERCOM_USART_INTFLAG_ERROR) {
+    else if (dev(uartnum)->INTFLAG.bit.ERROR) {
         /* clear error flag */
-        dev(uartnum)->INTFLAG.reg = SERCOM_USART_INTFLAG_ERROR;
+        dev(uartnum)->INTFLAG.reg |= SERCOM_USART_INTFLAG_ERROR;
```

Comment:
Please revert this change, not need for read/modify/write here, a simple write is sufficient: `Writing a one to this bit will clear the flag.`

Reply:
I guess you are right, I mis-interpreted the usage of it here and thought might be safer to read/write here - just as we do in other places.